### ISSUE TITLE
Allow partial supervision

### ISSUE THREAD

**Issue by LoicGrobol** [hopsparser/hopsparser#30]
So far, when training, we optimize for POS tagging, head-finding and dependency labelling, all at the same time. This works well when the gold corpus has all of these annotations but prevents us from training on corpora with partial annotations (POS but no syntax, heads but no deprel…). It would be useful to be allow partial supervision, where only part of these are taken into the loss.

Note: that can be done at the data processing level (just use the padding label for the parts we want to ignore) or at the parser level (with a parametrizable loss, in which case we might also want to add a weighting option or different loss combination options [but that's yet more hyperparameters…]).

This is somewhat connected to #9.

**Comment by dseddah** [comment]
that would be a good idea, especially if the data is corrupted or noisy and you want to predict only the noisy or missing parts. Eric de la CLergerie implemented something like that in his parser that we used for data driven treebank conversion (look at our lrec 2018 paper "Cheating a parser to death..."), that would be great to have something like this in an up to date parser such as this one.

**Comment by LoicGrobol** [comment]
I gave a look at this: it should not actually be that hard since we already ignore a word in the loss: the ROOT pseudo-token. We just need to give a way to ignore e.g. columns with `_` in the gold.

**Comment by LoicGrobol** [comment]
With release of v0.5.0, this is now the next I will be working on, hopefully it doesn't take a year.

**Comment by dseddah** [comment]
Cool :)  any news on the parsing graphs thingy (like enhanced ud and alikes) ?
